src/fastoad/_utils/pandas.py

```diff
@@ -0,0 +1,38 @@
+"""
+Module for pandas-related operations
+"""
+#  This file is part of FAST-OAD : A framework for rapid Overall Aircraft Design
+#  Copyright (C) 2024 ONERA & ISAE-SUPAERO
+#  FAST is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from typing import Callable, Optional
+
+import pandas as pd
+from packaging.version import Version
+
+# DataFrame.applymap() is deprecated since pandas 2.1.0, in favor of DataFrame.map()
+# We get the correct one here once and for all.
+if Version(pd.__version__) >= Version("2.1"):
+    MAP_METHOD = pd.DataFrame.map
+else:
+    MAP_METHOD = pd.DataFrame.applymap
```

Comment:
What makes you choose that way of doing things rather than setting pandas min version at 2.1 and only using applymap ?

Reply:
Actually, it's "map" that has to be used from pandas 2.1.
We could do as you say, but I just wanted to keep a broad compatibility range. Some user could have a reason to stick to pandas 1.x, and want to use FAST-OAD anyway...
Of course, we won't be able to keep compatibility with old packages indefinitely, but for this one, I thought it was worth the pain.
Maybe we'll need some day to keep track of these parts of code that ensure compatibility with old versions of Python or any library, to be sure to get rid of them when the support is no longer needed.
Surely, there is somewhere a tool that does already that...

Reply:
> Actually, it's "map" that has to be used from pandas 2.1.
Yes, sorry for the oversight
> Surely, there is somewhere a tool that does already that...
I'm sensing a sarcasm 😅

Reply:
Not at all a sarcasm. I didn't dig the subject, but generally in software development, when a problem is generic enough, someone(s) already proposed a solution.